analysis/src/CompletionFrontEnd.ml

```diff
@@ -1115,6 +1115,7 @@ let completionWithParser1 ~currentFile ~debug ~offset ~path ~posCursor ~text =
           processed := true
         | _ -> ());
       if not !processed then Ast_iterator.default_iterator.expr iterator expr
+      else inJsxContext := oldInJsxContext
```

Comment:
The only case that sets `inJsxContext`, case `Pexp_apply`, does not set `processed` to true.
So the `else` branch would not be executed.
Instead of just fixing the branch, what about restoring it unconditionally so it's future-proof.